Modules/Tracker/QuestieTracker.lua

```diff
@@ -569,88 +758,275 @@ function QuestieTracker:Update()
+                Questie:Error("|cff00bfffhttps://github.com/AeroScripts/QuestieDev/issues|r")
+            end
+            if qB.zoneOrSort > 0 then
+                qBZone = QuestieTracker.utils:GetZoneNameByID(qB.zoneOrSort)
+            elseif qB.zoneOrSort < 0 then
+                qBZone = QuestieTracker.utils:GetCatagoryNameByID(qB.zoneOrSort)
+            else
+                qBZone = qB.zoneOrSort
+                Questie:Error("SortID: |cffffbf00"..qB.zoneOrSort.."|r was not found in the Database. Please file a bugreport at:")
+                Questie:Error("|cff00bfffhttps://github.com/AeroScripts/QuestieDev/issues|r")
+            end
+            -- Sort by Zone then by Level to mimic QuestLog sorting
+            if qAZone == qBZone then
+                return qA.level < qB.level
+            else
+                if qAZone ~= nil and qBZone ~= nil then
+                    return qAZone < qBZone
+                else
+                    return qAZone and qBZone
+                end
+            end
+        end)
+    elseif Questie.db.global.trackerSortObjectives == "byProximity" then
+        table.sort(order, function(a, b)
+            local distanceA = GetDistanceToClosestObjective(a)
+            local distanceB = GetDistanceToClosestObjective(b)
+            local qA = QuestieDB:GetQuest(a)
+            local qB = QuestieDB:GetQuest(b)
+            local _, zoneA, _ = QuestieMap:GetNearestQuestSpawn(qA)
+            local _, zoneB, _ = QuestieMap:GetNearestQuestSpawn(qB)
+            local continent = getContinent(C_Map.GetBestMapForUnit("player"))
+            local continentA = getContinent(ZoneDataAreaIDToUiMapID[zoneA])
+            local continentB = getContinent(ZoneDataAreaIDToUiMapID[zoneB])
+
+            if ((continent == continentA) and (continent == continentB)) or ((continent ~= continentA) and (continent ~= continentB)) then
+                if distanceA == distanceB then
+                    return qA and qB and qA.level < qB.level;
+                end
+                if not distanceA and distanceB then
+                    return false;
+                elseif distanceA and not distanceB then
+                    return true;
+                end
+
+                return distanceA < distanceB;
+            elseif (continent == continentA) and (continent ~= continentB) then
+                return true
+            elseif (continent ~= continentA) and (continent == continentB) then
+                return false
+            end
+        end)
+
+        if not _QuestProximityTimer then
+            QuestieTracker:updateQuestProximityTimer()
+        end
     end
+
+    if (Questie.db.global.trackerSortObjectives ~= "byProximity") and _QuestProximityTimer and (_QuestProximityTimer:IsCancelled() ~= "true") then
+        _QuestProximityTimer:Cancel()
+        _QuestProximityTimer = nil
```

Comment:
```suggestion
    if (Questie.db.global.trackerSortObjectives ~= "byProximity") and questProximityTimer and (questProximityTimer:IsCancelled() ~= "true") then
        questProximityTimer:Cancel()
        questProximityTimer = nil
```

Is it really a string compare for `questProximity:IsCanceled()`?